app/main.py

```diff
@@ -1,2 +1,71 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list = [None] * 8
+
+    def __getitem__(
+            self, key: int | float | bool | str | tuple
+    ) -> None:
+        hash_key = hash(key)
+        index = hash_key % len(self.hash_table)
+
+        if (
+                self.hash_table[index]
+                and self.hash_table[index][2] == hash_key
+                and self.hash_table[index][0] == key
+        ):
+            return self.hash_table[index][1]
+        else:
+            for item in self.hash_table:
+                if item and item[0] == key:
+                    return item[1]
+
+        raise KeyError
+
+    def __setitem__(
+            self,
+            key: int | float | bool | str | tuple,
+            value: Any
```

Comment:
it's better to use Hashable typehint for the "key"

Reply:
Done